csharp/device/Microsoft.Azure.Devices.Client/DeviceClientHelper.cs

```diff
@@ -209,7 +351,14 @@ public Task<Message> ReceiveAsync()
             await this.EnsureOpenedAsync(false);
             await this.OnSendEventAsync(messages);
         }
+#endif
 
+#if NETMF
+        protected void EnsureOpened(bool explicitOpen)
+        {
+
```

Comment:
Again it's there just to keep 100% compatibility with the API.
Giving it some thought I really think it should be left there despite being empty. This way it a developer is working on a project that targets both platforms the code can be exactly the same thus achieving full code reuse. 
Maybe there should be a comment in these methods so that someone revisiting the code doesn't get the impression that something is missing.